Makefile

```diff
@@ -200,3 +182,21 @@ $(NODE_MODULES_TEST): package.json
 	env -u NODE_ENV npm prune
 
 .PHONY: all clean install devel-install dist node-cache rpm check vm update-po
+
+# checkout common files from Cockpit repository required to build this project;
+# this has no API stability guarantee, so check out a stable tag when you start
+# a new project, use the latest release, and update it from time to time
+COCKPIT_REPO_FILES = \
+	pkg/lib \
+	test/common \
+	$(NULL)
+
+COCKPIT_REPO_URL = https://github.com/cockpit-project/cockpit.git
+COCKPIT_REPO_SHA = d1a410cf9ccb33e983fbd38cad5a6fc95bc1c4b4 # 264-137-gd1a410cf9
+
+COCKPIT_REPO_STAMP = pkg/lib/cockpit-po-plugin.js
+$(COCKPIT_REPO_FILES): $(COCKPIT_REPO_STAMP)
+$(COCKPIT_REPO_STAMP): Makefile
+	@git rev-list --quiet --objects '$(strip $(COCKPIT_REPO_SHA))^{tree}' -- 2>/dev/null || \
```

Comment:
That strip looks weird, that's for cutting off the `#264 ...` comment? Let's just move the comment into a new line.

Reply:
What's going on here is that `make` parses the line up to the `#` — everything after is ignored.  The problem is that the space before the `#` gets included in the variable.  That's what's being stripped.

I prefer the one line thing.  If you're strongly against it I can change it, but I think it's the best way to do this, and the `$(strip)` thing isn't bad — in fact, I also applied it in some places where it wasn't strictly necessary.

Reply:
Not strongly, but it makes the already complicated code here even more unwieldy, plus it gets repeated a lot. Perhaps move it into a separate line/new variable?

Reply:
I added an extra variable and removed the 'extra' uses of `$(strip)` that were unnecessary (because the shell already absorbs the whitespace).